internal/oauth2/lazy_token.go

```diff
@@ -52,9 +52,9 @@ func (lt *LazyToken) acquireNewToken(ctx context.Context, forceRefresh bool) (st
 		lt.AccessToken = tokenRes.AccessToken
 		lt.Expiration = time.Now().Add(time.Duration(tokenRes.ExpiresIn) * time.Second)
 
-		logrus.WithContext(ctx).Infof("Acquired new token %s with expiration at: %s", lt.AccessToken, lt.Expiration)
+		logrus.WithContext(ctx).Infof("Acquired new token with expiration at: %s", lt.Expiration)
 	} else {
-		logrus.WithContext(ctx).Infof("AccessToken reused: %s (expires at %s)", lt.AccessToken, lt.Expiration)
+		logrus.WithContext(ctx).Infof("AccessToken reused, which expires at %s", lt.Expiration)
```

Comment:
Thanks for finding this!

Maybe adding a test like:
```go
diff --git a/internal/oauth2/lazy_token_test.go b/internal/oauth2/lazy_token_test.go
index 9e6e8300..70d82892 100644
--- a/internal/oauth2/lazy_token_test.go
+++ b/internal/oauth2/lazy_token_test.go
@@ -9,7 +9,11 @@ import (
        "testing"
        "time"
 
+       "github.com/stretchr/testify/assert"
        "github.com/stretchr/testify/require"
+
+       "github.com/sirupsen/logrus"
+       logrusTest "github.com/sirupsen/logrus/hooks/test"
 )
 
 func TestLazyToken(t *testing.T) {
@@ -34,6 +38,9 @@ func TestLazyToken(t *testing.T) {
        }))
        defer mockServer.Close()
 
+       _, hook := logrusTest.NewNullLogger()
+       logrus.AddHook(hook)
+
        clientID := "test-client-id"
        clientSecret := "test-client-secret"
        lazyToken := &LazyToken{
@@ -46,10 +53,21 @@ func TestLazyToken(t *testing.T) {
        require.NoError(t, err)
        require.Equal(t, "mock-token-1", token)
 
+       // ensure no token is not part of the logs
+       assert.Equal(t, 1, len(hook.Entries))
+       assert.Contains(t, hook.Entries[0].Message, "Acquired new token ")
+       assert.NotContains(t, hook.Entries[0].Message, "mock-token-1")
+       hook.Reset()
+
        token, err = lazyToken.Token(ctx)
        require.NoError(t, err)
        require.Equal(t, "mock-token-1", token)
 
+       assert.Equal(t, 1, len(hook.Entries))
+       assert.Contains(t, hook.Entries[0].Message, "AccessToken reused ")
+       assert.NotContains(t, hook.Entries[0].Message, "mock-token-1")
+       hook.Reset()
+
        // generates a new token when token expired
        lazyToken.Expiration = time.Now().Add(-time.Minute) // Expire the token
        token, err = lazyToken.Token(ctx)
@@ -61,6 +79,7 @@ func TestLazyToken(t *testing.T) {
        token, err = lazyToken.ForceRefresh(ctx)
        require.NoError(t, err)
        require.Equal(t, "mock-token-3", token)
+
 }
 
 func TestLazyTokenUnhappy(t *testing.T) {
diff --git a/vendor/modules.txt b/vendor/modules.txt
index fe12ac39..dcd9bddc 100644
--- a/vendor/modules.txt
+++ b/vendor/modules.txt
@@ -235,6 +235,7 @@ github.com/redhatinsights/platform-go-middlewares/logging/cloudwatch
 # github.com/sirupsen/logrus v1.9.3
 ## explicit; go 1.13
 github.com/sirupsen/logrus
+github.com/sirupsen/logrus/hooks/test
 # github.com/stretchr/objx v0.5.2
 ## explicit; go 1.20
 github.com/stretchr/objx

```
to ensure this does not regress (and that it is really fixed) woudl be beneficial here

Reply:
perfect! … taking it … :-D please check